tiledb/sm/cpp_api/map.h

```diff
@@ -1517,6 +1507,39 @@ class Map {
 /*            DEFINITIONS            */
 /* ********************************* */
 
+template <typename T>
+T MapItem::get(const std::string& attr) const {
+  using DataT = typename impl::TypeHandler<T>;
+
+  const Context& ctx = ctx_.get();
+
+  tiledb_kv_item_t* item;
+  const void* key;
+  tiledb_datatype_t type;
+  uint64_t size;
+
+  ctx.handle_error(
+      tiledb_kv_item_get_key(ctx, item_.get(), &key, &type, &size));
+
+  ctx.handle_error(
+      tiledb_kv_get_item(ctx, map_->ptr().get(), key, type, size, &item));
+
+  typename DataT::value_type* vdata;
+
+  ctx.handle_error(tiledb_kv_item_get_value(
+      ctx, item, attr.c_str(), (const void**)&vdata, &type, &size));
+
+  auto num = static_cast<unsigned>(size / sizeof(typename DataT::value_type));
+  impl::type_check<T>(type);  // Just check type
```

Comment:
also here, a workaround would be to have item as a std::uniqute_ptr<> and specify a custom deleter, that way RAII cleans it up